spacialgaze-plugins/news.js

```diff
@@ -0,0 +1,87 @@
+/*
+* News System for SpacialGaze
+* Credits: Lord Haji
+*/
+
+'use strict';
+
+let moment = require('moment');
+
+function generateNews() {
+	let newsData, newsDisplay = [];
+	Object.keys(Db('news').object()).forEach(announcement => {
+		newsData = Db('news').get(announcement);
+		newsDisplay.push(`<h4>${announcement}</h4>${newsData[1]}<br /><br />—${SG.nameColor(newsData[0], true)} <small>on ${newsData[2]}</small>`);
+	});
+	return newsDisplay;
+}
+
+function hasSubscribed(user) {
+	if (typeof user === 'object') user = user.userid;
+	if (Db('NewsUnsubscribers').get(toId(user)) === 1) return false;
+	return true;
+}
+
+SG.showNews = function (user) {
+	if (!Users(user)) return false;
+	let newsDisplay = generateNews();
+	if (newsDisplay.length === 0) return false;
+	if (!hasSubscribed(user)) return false;
+	if (newsDisplay.length > 0) {
+		newsDisplay = newsDisplay.join('<hr>');
+		return Users(user).send(`|pm| SpacialGaze News|${Users(user).getIdentity()}|/raw ${newsDisplay}`);
+	}
+};
+
+exports.commands = {
+	news: 'serverannouncements',
+	announcements: 'serverannouncements',
+	serverannouncements: {
+		'': 'view',
+		display: 'view',
+		view: function (target, room, user) {
+			return user.send('|popup||wide||html|' +
+				"<center><strong>Current server announcements:</strong></center>" +
+					generateNews().join('<hr>')
+			);
+		},
+		remove: 'delete',
+		delete: function (target, room, user) {
+			if (!this.can('ban')) return false;
+			if (!target) return this.parse('/help serverannouncements');
+			if (!Db('news').has(target)) return this.errorReply("News doesn't exitst");
+			Db('news').delete(target);
+			this.privateModCommand(`(${user.name} deleted server announcement titled: ${target}.)`);
+		},
+		add: function (target, room, user) {
+			if (!this.can('ban')) return false;
+			if (!target) return this.parse('/help serverannouncements');
+			let parts = target.split(',');
+			if (parts.length !== 2) return this.errorReply("Usage: /news add [title], [desc]");
+			let title = parts[0], desc = parts[1], postedBy = user.name;
+			let postTime = moment(Date.now()).format("MMM D, YYYY");
```

Comment:
Moment is depreciated, use `Chat.toDurationString()` instead. https://github.com/HoeenCoder/SpacialGaze/blob/master/chat.js#L978